flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java

```diff
@@ -283,6 +283,25 @@ public void testTransformationSetParallelism() {
         assertThat(vertices.get(2).isParallelismConfigured()).isTrue();
     }
 
+    @Test
+    public void testTransformationSetMaxParallelism() {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        /* The max parallelism of the environment (that is inherited by the source)
+        and the parallelism of the map operator needs to be different for this test */
+        env.setMaxParallelism(4);
+        env.fromSequence(1L, 3L).map(i -> i).setMaxParallelism(10).print().setMaxParallelism(20);
```

Comment:
It's better to introduce variables for each DataStream/DataStreamSink and then get StreamNode by their ID.

```java
DataStreamSource<Long> source = env.fromSequence(1L, 3L);
assertThat(streamGraph.getStreamNode(source.getId()).getMaxParallelism()).isEqualTo(4);
```

Reply:
Oh funny, DataStreamSink doesn't have `getId()` as well, I can add that to this PR or create a new issue.